Task 12

yevhenorlov opened the pull request:
🍿 [demo](https://yevhenorlov.com/assignments_yevhenorlov/task_12/build/index.html)

📝 [source](https://github.com/kottans-frontend-2018/assignments_yevhenorlov/tree/task-12/task_12)

Thanks in advance!

yevhenorlov committed:
add boilerplate

yevhenorlov committed:
add markup

yevhenorlov committed:
add css

yevhenorlov committed:
fix grid-rows in pizza to support safari

yevhenorlov committed:
update icons

yevhenorlov committed:
clean up useless code

yevhenorlov committed:
refactor button hover

yevhenorlov committed:
clean up, add tile image to footer/header background

yevhenorlov committed:
fix footer font-weight

yevhenorlov committed:
add display:grid to body element

yevhenorlov committed:
add task_12 links to readme

viktor-yakubiv commented on task_12/html-dev/index.html:
```diff
@@ -0,0 +1,136 @@
+      </div>
+      <div class="pizza">
+        <img src="assets/img/pizza6.jpg" alt="pizza6" class="pizza--picture">
+        <time class="pizza--timestamp">10:56:22</time>
```
It's better to use `datetime` attribute with time tag.

Time element developed to determine actual time when displaying relative one, i. e. we may have displayed text _yesterday_, but it was actually _2018-01-27 16:03:18_.

---

Learn BEM more deeply. _M_ stands for modifier. Modifiers represent state changes of a block or an element. Examples:
- `card` — a container, i.e. a block
- `card__content` child element of `card`, i.e. an element or an inner block
- `card--opened` — modifier for `card`
- `card__content--empty` — empty `card__content` element, modifier

yevhenorlov replied:
Well, that's embarrassing — I totally mixed up the order of dashes and underscores :) Thought it was `block--element__modifier`. 

Thanks for pointing it out, and thanks for the review. I think I'll fix the code tomorrow and add a commit to this PR.

viktor-yakubiv commented on task_12/gulpfile.js:
```diff
@@ -0,0 +1,155 @@
+
+//js minify example
+// to run "gulp index"
+// gulp.task('index', function () {
```
Avoid commented code in the repository. We have git history for it. Otherwise you may not to have this code and may delete it.

Keep your repository clean.

viktor-yakubiv reviewed (approved):
Looks good. Great job!

You may merge.

If you want to you may improve resolving comments in the code, or send them using another PR.

OleksiyRudenko commented:
Great job!

This may come from css grid support specifics, but here is what I see in Chrome 63.0.3239.84 (64-bit, Windows 10): the rendered content is a bit wider than the port view.

![sshot-180128-yo](https://user-images.githubusercontent.com/8686631/35486498-7beeeacc-0477-11e8-86c0-de2e7d683d51.png)

Setting `header { width: 100%; }` (instead of `100vw`) seems to fix the issue.

yevhenorlov committed:
fix BEM format, remove useless code

yevhenorlov committed:
increase wrapper padding

yevhenorlov commented:
Thanks for your comments!

@viktor-yakubiv, I've fixed the BEM issue and removed commented code, and I'll keep in mind your tip about `datetime` attribute for the next stage of the task - seems pointless to change it now, before there's any application logic in place. Appreciate your feedback 👍

@OleksiyRudenko, good catch. However, looks like there's an irritating issue with 100% width of the content on MacOS - when the scrollbar gets transparent, this silly gap appears:

![screen shot 2018-01-29 at 10 18 34 am](https://user-images.githubusercontent.com/17388747/35500654-e190c41e-04df-11e8-9a5b-7de4aec4bf64.png)

So I've decided to keep the `width: 100vw`and simply increase the padding from 10px to 20px, should work ok now. Thanks!
